torch/distributed/distributed_c10d.py

```diff
@@ -1238,38 +1229,21 @@ def new_group(ranks=None, timeout=_default_pg_timeout):
         else:
             group_rank = None
     else:
-        input_ranks = []
         ranks = list(range(global_world_size))
         group_world_size = global_world_size
         group_rank = global_rank
 
-    if default_backend == Backend.MPI:
-        in_group = global_rank in ranks
-        pg = _new_process_group_helper(group_world_size,
-                                       group_rank,
-                                       input_ranks,
-                                       in_group,
-                                       group_name,
-                                       timeout=timeout)
-    else:
-        # Release ranks not in the group
-        if global_rank not in ranks:
-            return GroupMember.NON_GROUP_MEMBER
-
-        if default_backend != Backend.MPI:
-            pg = _new_process_group_helper(group_world_size,
-                                           group_rank,
-                                           input_ranks,
-                                           True,
-                                           group_name,
-                                           timeout=timeout)
+    pg = _new_process_group_helper(group_world_size,
+                                   group_rank,
```

Comment:
IIUC, MPI ignores this `group_rank`, right?

Reply:
Yes, it only uses the `group_ranks` list. Then it uses that to figure out itself what its own rank in that group is. This is a side effect of how you create new groups in MPI.